core/shared/src/main/scala/cats/parse/LocationMap.scala

```diff
@@ -92,13 +99,17 @@ class LocationMap(val input: String) {
         // so we are pointing into a row
         val rowStart = firstPos(row)
         val col = offset - rowStart
-        Some((row, col))
+        Caret(offset, row, col)
       } else {
         // idx is exactly the right value because offset is beginning of a line
-        Some((idx, 0))
+        Caret(offset, idx, 0)
       }
     }
 
+  def toCaret(offset: Int): Option[Caret] =
+    if (offset < 0 || offset > input.length) None
```

Comment:
:art: this condition is popping a few times would a `isValidOffset` be worth?

Reply:
good idea. I've made that change (and avoid checking it so many times).